File: backend/internal/model/department.go
package model

type Department struct {
	SnowflakeTenantModel
	Name     string `gorm:"size:100;not null" json:"name"`
	ParentID int64  `gorm:"index;default:0" json:"parent_id"`
	Sort     int    `gorm:"default:0" json:"sort"`
	Status   int    `gorm:"default:1" json:"status"`
	LeaderID int64  `gorm:"default:0" json:"leader_id"`
	Children []Department `gorm:"-" json:"children,omitempty"`
}

type UserDepartment struct {
	SnowflakeTenantModel
	UserID int64 `gorm:"index;not null" json:"user_id"`
	DeptID int64 `gorm:"index;not null" json:"dept_id"`
}

type RoleDept struct {
	SnowflakeTenantModel
	RoleID int64 `gorm:"index;not null" json:"role_id"`
	DeptID int64 `gorm:"index;not null" json:"dept_id"`
}

func (Department) TableName() string {
	return "departments"
}

func (UserDepartment) TableName() string {
	return "user_departments"
}

func (RoleDept) TableName() string {
	return "role_depts"
}

File: web/src/api/department.ts
import request, { type ApiResponse } from './request'

export interface Department {
  id: number
  name: string
  parent_id: number
  sort: number
  status: number
  leader_id: number
  children?: Department[]
  created_at: string
  updated_at: string
}

export interface CreateDeptParams {
  name: string
  parent_id?: number
  sort?: number
  leader_id?: number
}

export interface UpdateDeptParams {
  name?: string
  parent_id?: number
  sort?: number
  status?: number
  leader_id?: number
}

const deptApi = {
  getTree(): Promise<ApiResponse<Department[]>> {
    return request.get('/departments/tree')
  },

  create(params: CreateDeptParams): Promise<ApiResponse<Department>> {
    return request.post('/departments', params)
  },

  update(id: number, params: UpdateDeptParams): Promise<ApiResponse<Department>> {
    return request.put(`/departments/${id}`, params)
  },

  delete(id: number): Promise<ApiResponse<null>> {
    return request.delete(`/departments/${id}`)
  },

  assignUser(userId: number, deptId: number): Promise<ApiResponse<null>> {
    return request.post(`/departments/${deptId}/users/${userId}`)
  },

  removeUser(userId: number, deptId: number): Promise<ApiResponse<null>> {
    return request.delete(`/departments/${deptId}/users/${userId}`)
  }
}

export default deptApi
